doc/in-progress/package-versioning.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Package Versioning
+---
+
+Sui smart contracts are organized into [upgradeable](./package-upgrades.md) packages and, as a result, multiple versions of any given package can exist on-chain. Before a package can be used on the chain, it is [published](move/debug-publish.md#publishing-a-package) at its first (a.k.a, original) version. A new version of a package is created whenever the package is [upgraded](./package-upgrades.md), and an upgrade always happens with respect to the version of the package immediately preceding the upgraded one in the versions history. In other words, an upgrade to the Nth version of a package is always based on the N-1th version of the package (this is enforced by the system) and one can never create the Nth version of the package based on a version that's smaller than N-1th version. For example a package can be upgraded from version 1 to 2 but once this happens only an upgrade from version 2 to 3 is possible and an upgrade from version 1 to 3 is forbidden.
+
+Note that there is also a notion of versioning present in the [manifest](./move/manifest.md) files. You can observe it both in the [package section](./move/manifest.md#package-section) and in the [dependencies section](./move/manifest.md#dependencies-section). For example:
+```toml
+[package]
+name = "some_pkg"
+version = "1.0.0"
+
+[dependencies]
+another_pkg = { git = "https://github.com/another_pkg/another_pkg.git" , version = "2.0.0"}
+```
+These fields in the manifest file, however, at this point are only used for user-level documentation purposes and are ignored by the publish and upgrade commands. Another way of looking at it is that if a developer publishes a package with a certain package version in the manifest file and then modifies and re-publishes the same package with a different version (using publish command rather than upgrade command), these two packages will be considered different packages, rather than on-chain versions of the same package. For example, none of these packages should be used as a [dependency override](./dependency-overrides.md) to stand in for the other one - while this kind of override is possibly to specify when building a package, it will result in an error when publishing/upgrading on-chain.
+
+## Compatibility
+
+Sui comes with a set of built-in package upgrade policies, listed here in the order of strictness:
+
+- immutable - this policy is a bit of a misnomer as having it applied means that the package cannot be upgraded at all
```

Comment:
nit: I think it is a misnomer to call it a compatibility policy because it's the absence of an `UpgradeCap` rather than `immutable` being a misnomer because it means that the package cannot be upgraded at all -- in that sense the name is consistent.